client/ayon_blender/plugins/load/load_cache.py

```diff
@@ -52,15 +53,15 @@ def _update_transform_cache_path(self, asset_group, libpath, prev_filename):
 
             modifier = obj.modifiers.new(
                 name='MeshSequenceCache', type='MESH_SEQUENCE_CACHE')
-            bpy.ops.cachefile.open(filepath=libpath.as_posix())
             modifier.cache_file = bpy.data.cache_files[-1]
-            modifier.cache_file.name = os.path.basename(libpath.as_posix())
+            cache_file_name = os.path.basename(libpath.as_posix())
+            modifier.cache_file.name = cache_file_name
             modifier.cache_file.filepath = libpath.as_posix()
             modifier.cache_file.scale = 1.0
             bpy.context.evaluated_depsgraph_get()
 
             modifier.object_path = (
-                modifier.cache_file.object_paths[0].path)
+                modifier.cache_file.object_paths[i].path)
```

Comment:
Something tells me this update will only work if the amount of objects in the alembic file would not change?

Reply:
it will still change, the object path means the hierarchy object shapes inside the alembic file.

Reply:
Sure - but if there are more objects in the alembic than in the asset group currently - it wouldn't have those new objects? Or if the are now less objects in the alembic than in the asset group this would error due to it being out of the length of the object paths array?

Reply:
> Sure - but if there are more objects in the alembic than in the asset group currently - it wouldn't have those new objects? Or if the are now less objects in the alembic than in the asset group this would error due to it being out of the length of the object paths array?

can you think of better way to deal with it? If we always used the first object path it would end up the scenario like [this](https://github.com/ynput/ayon-blender/pull/18#pullrequestreview-2247620541)

Reply:
The best way I can think of is:

- Before update build a lookup of current scene objects from their object paths in the alembic.
- Then after update:
    - Update those that existed before and still exist.
    - Delete those scene objects that do not exist in the new state after update.
    - Create the scene objects that do exist after the update but didn't exist before.

This has the caveat that you'd lose any local scene changes on nodes that are removed on updates, etc. but at least your scene contents actually match what the alembic contains?

Reply:
> The best way I can think of is:
> 
> * Before update build a lookup of current scene objects from their object paths in the alembic.
> * Then after update:
>   
>   * Update those that existed before and still exist.
>   * Delete those scene objects that do not exist in the new state after update.
>   * Create the scene objects that do exist after the update but didn't exist before.
> 
> This has the caveat that you'd lose any local scene changes on nodes that are removed on updates, etc. but at least your scene contents actually match what the alembic contains?

I tested with multiple scenarios when setting version to more or less object paths in the next version than in the previous version. It still reads as expected. But we need to be concerned about how to make it right. It's a bit struggling when trying to implement your approach above, knowing that we need to get the object path from the cache file from the modifier, which is inherited from the object. And comparing it with the object paths from the "removed" modifier (Maybe I did something wrong). It could be some limitation of the mesh sequence modifiers, of which we can't get the object path globally.  The snippet below showed how I got the modifier(but still fails to get the correct object path, unlike the current commit):
```python
    def _update_transform_cache_path(self, asset_group, libpath, prev_filename):
        """search and update path in the transform cache modifier
        If there is no transform cache modifier, it will create one
        to update the filepath of the alembic.
        """
        bpy.ops.cachefile.open(filepath=libpath.as_posix())
        old_object_paths = []
        for i, obj in enumerate(asset_group.children):
            names = [modifier.name for modifier in obj.modifiers
                     if modifier.type == "MESH_SEQUENCE_CACHE"]
            file_list = [file for file in bpy.data.cache_files
                         if file.name.startswith(prev_filename)]
            if names:
                for name in names:
                    modifier = obj.modifiers.get(name)
                    if modifier.cache_file is not None:
                        for obj_path in modifier.cache_file.object_paths:
                            obj_path_name = obj_path.path
                            if obj_path_name in old_object_paths:
                                continue
                            old_object_paths.append(obj_path_name)
                    obj.modifiers.remove(modifier)
            if file_list:
                bpy.data.batch_remove(file_list)

            modifier = obj.modifiers.new(
                name='MeshSequenceCache', type='MESH_SEQUENCE_CACHE')
            modifier.cache_file = bpy.data.cache_files[-1]
            cache_file_name = os.path.basename(libpath.as_posix())
            modifier.cache_file.name = cache_file_name
            modifier.cache_file.filepath = libpath.as_posix()
            modifier.cache_file.scale = 1.0
            bpy.context.evaluated_depsgraph_get()

            modifier.object_path = (
                modifier.cache_file.object_paths[i].path)

        return libpath
```

Reply:
Admittedly I know too little of blender to quickly help here API wise. If needed I can put in time and see if I can reproduce and help out code-wise. But it may take me some time and I'm not sure if my time is best spent there now, sorry.